### ISSUE TITLE
Does buildpack support replace directive in go.mod?

### ISSUE THREAD

**Issue by yurifedoseev** [GoogleCloudPlatform/buildpacks#285]
Hi!
I'm working on a project with several google cloud functions written in go. Currently we deliver dependencies uploading `vendor` directory. I want to switch to `go.mod` approach, which looks more reliable.
The problem we use fork of `firestore` client which is define in `go.mod` via `replace` directive:
`replace cloud.google.com/go/firestore => github.com/laigames/google-cloud-go/firestore v1.9.0-1.laigames`

With `go.mod` approach my gcp build failed because new code from fork wasn't not found. Looks like `replace` directive was ignored.
I'm curious is `replace` directive supported in buildpack or there are some limitations?

**Comment by smsohan** [comment]
Can you please share what you saw in the failed build output? Or possibly a minimal repro that we can use to test?

**Comment by yurifedoseev** [comment]
Thank you for the explanation!

> One workaround is to keep both your go.mod and to vendor your dependencies using the go command before deploying: https://cloud.google.com/functions/docs/writing/specifying-dependencies-go#specify_dependencies_with_a_vendor_directory

I'm currently using this approach by uploading `vendor` folder after `go mod vendor` command. 
The problem is that `vendor` approach is quite hacky.
I have to apply this workaround for `metadata/context` https://github.com/googleapis/google-cloud-go/issues/1947

The previous workaround doesn't allow to fix that warning (because removing dependency breaks the build)
`WARNING: Your vendored dependencies do not contain the functions framework (github.com/GoogleCloudPlatform/functions-framework-go/funcframework). If there are conflicts between the vendored packages and the dependencies of the framework, you may encounter unexpected issues.`

I want eventually move to `go.mod` approach but it looks the only way is to remove any `replace` directive

**Comment by jama22** [comment]
Closing because it sounds like you were unblocked with the workaround suggested.

It does sound like there's a feature request somewhere here around better support for vendored dependencies in go functions. @yurifedoseev feel free to create a new issue as a feature request for what your ideal use case is. Thanks
